Automattic/wp-calypso
# Domains: Improve all domains page performance

### Changes proposed in this Pull Request

When you have more than a handful of domains, the "all domains" page can take quite a while to load. This PR is an initial effort to improve the performance of that page. Up until now, when you loaded the `/domains/manage` page, we made one call o `all-domains` and one call to `site/%s/domains` for each site the user has.

This PR replaces the individual calls to `site/%s/domains` by a single call to the new `site-domains-in-bulk` endpoint, which retrieves all necessary domain data for all the user's sites at once.

Further improvements:
- We probably don't need `QueryAllDomains` in `all-domains` anymore

### Testing instructions

This has to be tested with D78897-code.

- Build this branch locally or open the live Calypso link
- Go to `/domain/manage?flags=force-sympathy` (the `force-sympathy` flag forces reloading data from the server)
- Ensure all your domains have the correct status and notices
